nucypher/blockchain/eth/signers/__init__.py

```diff
@@ -0,0 +1,29 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from nucypher.blockchain.eth.signers.base import Signer
+from nucypher.blockchain.eth.signers.software import ClefSigner, KeystoreSigner
+from nucypher.blockchain.eth.signers.hardware import TrezorSigner
+
+
+SIGNERS = {
+    ClefSigner.URI_SCHEME: ClefSigner,
+    KeystoreSigner.URI_SCHEME: KeystoreSigner,
+    TrezorSigner.URI_SCHEME: TrezorSigner
+}
+
+Signer.SIGNERS = SIGNERS
```

Comment:
Good eye - The only reason I ended up with this here is to avoid importing Issues.  Where can can it be defined with all of the subclass names still in scope?

Reply:
You're right, circular import will be a problem. I suppose the only alternative would be to import the derived classes dynamically, and have a method of `Signer` like 
```
    @staticmethod
    def signers():
        from nucypher.blockchain.eth.signers.software import ClefSigner, KeystoreSigner
        from nucypher.blockchain.eth.signers.hardware import TrezorSigner
        return {
            ClefSigner.URI_SCHEME: ClefSigner,
            KeystoreSigner.URI_SCHEME: KeystoreSigner,
            TrezorSigner.URI_SCHEME: TrezorSigner,
        }
```

Reply:
Yeah - It's just a matter of namespace - might be able to do some kind of dynamic class creation, or subclass_init hook, but that might overkill.

Reply:
Wouldn't the code from my comment be enough?

Reply:
Yes - it will work - but the namespace will be `signers.Signer.signers()` which I find less useful than simply `signers.SIGNERS`.

Reply:
Does it ever get used like that? Currently all the use cases are `cls.SIGNERS`.

(In fact, if it stays that way, the constant's name in `__init__.py` should perhaps be changed to `_SIGNERS`)

Reply:
:heavy_check_mark: - Internalized.

Reply:
Cool, removing the module constant completely is a logical next step.

So, why are you reluctant to make it a method in the `Signer` class? Is it a performance issue, dislike of dynamic imports, or do you want its usage to look as a constant?

Reply:
I'm *almost* sold on moving it. :smile: 

Yeah - I think mostly the last two: dislike of dynamic imports, and usage as a constant.